lib/inv_multi_cg_quda.cpp

```diff
@@ -440,15 +454,21 @@ namespace quda {
       }
       num_offset_now -= converged;
 
+      // exit early so that we can finish of shift 0 using CG and allowing for mixed precison refinement
+      if(num_offset_now==1){
+        exitearly=true;
+        num_offset_now--;
+      }
+
       // this ensure we do the update on any shifted systems that
       // happen to converge when the un-shifted system converges
-      if ( (convergence(r2, stop, num_offset_now) ||  k == param.maxiter) && aux_update == true) {
+      if ( (convergence(r2, stop, num_offset_now) || exitearly|| k == param.maxiter) && aux_update == true) {
 	if (getVerbosity() >= QUDA_VERBOSE) 
 	  printfQuda("Convergence of unshifted system so trigger shiftUpdate\n");
 	
 	// set worker to do all updates at once
 	shift_update.updateNupdate(1);
-	shift_update.apply(0);
+	shift_update.apply(nullptr);
```

Comment:
Since the function signature is `void apply(const cudaStream &stream)` I'm not sure if passing a `nullptr` is syntactically what we should be doing.  It works of course, since a `cudaStream_t` is defined to be a pointer
```c++
typedef CUstream_st * cudaStream_t
```
and that definition won't change.  I don't care that strongly, but it seems funny to make this change here where `0` is used everywhere else in QUDA.